python/cudf/cudf/core/reshape.py

```diff
@@ -278,9 +327,25 @@ def concat(objs, axis=0, ignore_index=False, sort=None):
                 result = objs[0].copy()
             return result
         else:
-            return cudf.DataFrame._concat(
-                objs, axis=axis, ignore_index=ignore_index, sort=sort
+            result = cudf.DataFrame._concat(
+                objs,
+                axis=axis,
+                join=join,
+                ignore_index=ignore_index,
+                sort=sort,
```

Comment:
If we have a mix of multiple non-empty and empty dataframes , we'll erroneously branch into this logic. We shouldn't be branching into this code. Instead, we'll have to track after line 313 by checking if we have matching columns in the empty dataframes and then decide which code-flow to go through.

Here is the example:
```python
>>> df
    a     b
p  10   sds
o  11  sds1
>>> empty_df
Empty DataFrame
Columns: []
Index: []

>>> pd.concat([df.to_pandas(), empty_df.to_pandas(), df.to_pandas(), empty_df.to_pandas()], join='inner', ignore_index=True)
Empty DataFrame
Columns: []
Index: [0, 1, 2, 3]
>>> pd.concat([df.to_pandas(), empty_df.to_pandas(), df.to_pandas(), empty_df.to_pandas()], join='inner', ignore_index=False)
Empty DataFrame
Columns: []
Index: [p, o, p, o]



>>> cudf.concat([df, empty_df, df, empty_df], join='inner', ignore_index=True)
    a     b
0  10   sds
1  11  sds1
2  10   sds
3  11  sds1
>>> cudf.concat([df, empty_df, df, empty_df], join='inner', ignore_index=False)
    a     b
p  10   sds
o  11  sds1
p  10   sds
o  11  sds1
```